### PR TITLE
Settings rework for new web UI + xp/money% in spoiler log

### PR DESCRIPTION
**Pull request by Cyb3RGER**
this rework includes:
- reordering and grouping of options and args
- changed description for options 
- added description for difficulty
- presets for web UI

also I snuck in a commit for xp/money% in the spoiler log because why is that not a thing yet :P

### CODE REVIEW

File: main.c [hunk 1]
```diff
@@ -1922,6 +1974,7 @@ int main(int argc, const char** argv)
         fprintf(flog,"Placement from file" ENDL);
     else
         fprintf(flog,"Tree depth: %d, cyber logic score: %d, cyber gameplay score: %d%s", treedepth, cyberlogicscore, cybergameplayscore, ENDL);
+    fprintf(flog,"Exp%%: %d, Money%%: %d%s", u8_fraction_to_percent(exp_num,exp_den), u8_fraction_to_percent(money_num,money_den), ENDL);
```
Comment (black-sliver) [inline main.c:1978]: you sure about `Money%: 100` instead of `Money: 100%`?

File: main.c [hunk 2]
```diff
@@ -238,6 +247,19 @@ enum option_indices {
         *t++ = 0;\
     } while (false)
 #endif
+
+struct preset {const char* name; const char* settings; int exp; int money;};
```
Comment (black-sliver) [inline main.c:251]: we could add `--presets` and `--preset <name>` to the CLI version to match features

File: main.c [hunk 3]
```diff
@@ -238,6 +247,19 @@ enum option_indices {
...
+const static struct preset presets[] = {
+    {"First-Timer", "rel", 300, 300},
+    {"Beginner", "rel", 200, 250},
+    {"Advanced", "rlABGSCd67", 150, 200},
```
Comment (black-sliver) [inline main.c:255]: spoiler log `on` by default for advanced/pro/hell/hell+? Also not sure hell+ is a good idea :D

File: main.c [hunk 4]
```diff
@@ -125,55 +125,64 @@ const char* POOL_STRATEGY_VALUES[] = { "Balance", "Random", "Bosses", NULL };
...
 
 #define DEFAULT_difficulty 1
-struct option { char key; uint8_t def; const char* text; const char* info; const char* description; const char** state_names; };
+struct option { char key; uint8_t def; const char* text; const char* info; const char* description; const char** state_names; const char* section; const char* subsection;};
 const static struct option options[] = {
-    { 0,   1, "Open World", NULL,          "Make windwalker available in every firepit", OFF_ON },
-    { 'k', 1, "Keep dog", NULL,            "Keep dog in some places to avoid softlocks", OFF_ON },
-    { '1', 1, "Fix sequence", NULL,        "Fix some sequence breaks: Volcano rock, final boss hatch", OFF_ON },
-    { '2', 1, "Fix cheats", NULL,          "Fix vanilla cheats: Infinite call beads", OFF_ON },
+    { 0,   1, "Open World", NULL,          "Make windwalker available in every firepit", OFF_ON, NULL },
```
Comment (black-sliver) [inline main.c]: it think this is missing `, NULL` for the subsection

File: main.c (hunk 1 above)
Comment (Cyb3RGER) [reply main.c:1978]: tbh, I am fine with both. Just think `Money%: 100` is easy to spot in the log and matches the input fields.

File: main.c (hunk 2 above)
Comment (Cyb3RGER) [reply main.c:251]: if by we, you mean, you, then I agree :P. I don't think many people use the CLI version and this is just QoL-Feature so I would not see this as a high priority.

File: main.c (hunk 3 above)
Comment (Cyb3RGER) [reply main.c:255]: tbh I don't really see any reason to not generate a log file unless it's specifically a race but I can also remove it for those. your call.
hell+ is for the Menblocks of this world. also is just wanted a 1% preset :D

Comment (black-sliver) [reply main.c:255]: maybe let's call it menblock for the memes?

File: main.c (hunk 2 above)
Comment (black-sliver) [reply main.c:251]: if presets are not supported by the CLI version, then `--presets.json` should not be compiled in:
* remove from help
* `#ifdef` away in the option parsing (so you get an error specifying it)
* `#ifdef` away the implementation (so you don't get unused function warning)

**Pull request merged**